cpp/src/arrow/util/bit-util.h

```diff
@@ -686,6 +690,65 @@ void GenerateBitsUnrolled(uint8_t* bitmap, int64_t start_offset, int64_t length,
   }
 }
 
+// A function that visits each bit in a bitmap and calls a visitor function with a
+// boolean representation of that bit. This is intended to be analogous to
+// GenerateBits.
+template <class Visitor>
+void VisitBits(const uint8_t* bitmap, int64_t start_offset, int64_t length,
+               Visitor&& visit) {
+  for (int64_t index = start_offset; index < start_offset + length; ++index) {
+    visit(arrow::BitUtil::GetBit(bitmap, index));
```

Comment:
Have you benchmarked this? BitmapReader should probably be a bit faster.

Reply:
Thank you for taking a look at this @pitrou. I have been running some benchmarks of `VisitBits` and `VisitBitsUnrolled` at the `featherread` layer, and I can try benchmarking `BitmapReader` too.

Do you have a recommendation on the best way to attach images to a Github conversation? I'd like to upload these benchmark plots that I generated.

Reply:
I meant try to implement `VisitBits` using `BitmapReader`.
As for attaching plots, I don't know. You may try one of the image hosting services around, e.g. https://framapic.org/

Reply:
I replaced the usage of  `GetBit` in the `VisitBits` method by `BitmapReader` as a part of [this commit](https://github.com/apache/arrow/pull/4328/commits/35143c22c9d005cfc32cc79b01d081cd5b31944f). We are seeing a consistent slight improvement in performance (~3% speedup) when using `BitmapReader` here compared to using `GetBit` directly. 

@pitrou Is this in line with what you'd expect?

Reply:
@rdmello This is good indeed, thank you!